backup-loop.sh

```diff
@@ -1,25 +1,57 @@
 #!/bin/bash
 
+set -eu
+
 log() {
-  level=$1
+  if [ "$#" -lt 2 ]; then
+    echo "Wrong number of arguments passed to log function" >&2
+    exit 1
+  fi
+  local level=${1}
   shift
   echo "$(date -Iseconds) ${level} $*"
 }
 
-backupSet=${BACKUP_SET}
-excludes="--exclude '*.jar'"
+find_old_backups() {
+  find "${DEST_DIR}" -mtime "+${PRUNE_BACKUPS_DAYS}" "${@}"
+}
+
+backupSet="${BACKUP_SET:-}"
+# shellcheck disable=SC2089
+excludes="${EXCLUDES:-"--exclude '*.jar'"}"
 
-case $TYPE in
+: "${SRC_DIR:=/data}"
+: "${DEST_DIR:=/backups}"
+: "${BACKUP_NAME:=world}"
+: "${INITIAL_DELAY:=120}"
+: "${INTERVAL_SEC:=86400}"
+: "${PRUNE_BACKUPS_DAYS:=7}"
+: "${TYPE:=VANILLA}"
+: "${RCON_PORT:=25575}"
+: "${RCON_PASSWORD:=minecraft}"
+
+case "${TYPE}" in
   FTB|CURSEFORGE)
-    cd ${SRC_DIR}/FeedTheBeast
+    cd "${SRC_DIR}/FeedTheBeast"
+    ;;
+  BUKKIT|SPIGOT|PAPER)
+    cd "${SRC_DIR}"
+    if [ -z "${LEVEL:-}" ]; then
+      LEVEL="world world_nether world_the_end"
+    fi
     ;;
   *)
-    cd ${SRC_DIR}
+    cd "${SRC_DIR}"
     ;;
 esac
 
+: "${LEVEL:=world}"
+
+log INFO "waiting initial delay of ${INITIAL_DELAY} seconds..."
```

Comment:
Should we display it in original form or already calculated? (see line below with sleep)

Reply:
I guess I don't follow what the difference would be.

Reply:
Hopefully explained in more in-depth: https://github.com/itzg/docker-mc-backup/pull/2#discussion_r303038974
Original: `2019-07-12T17:40:35+02:00 INFO waiting initial delay of 2*60 seconds...`
Calculated: `2019-07-12T17:40:35+02:00 INFO waiting initial delay of 120 seconds...`